platform/darwin/src/MGLOfflinePack.mm

```diff
@@ -147,11 +147,16 @@ - (void)reloadWithCompletionHandler:(void (^)(NSError * _Nullable error))complet
     MGLOfflineStorage *sharedOfflineStorage = [MGLOfflineStorage sharedOfflineStorage];
     __weak MGLOfflinePack *weakSelf = self;
     [sharedOfflineStorage getPacksWithCompletionHandler:^(NSArray<MGLOfflinePack *> *packs, __unused NSError * _Nullable error) {
+        MGLOfflinePack *strongSelf = weakSelf;
         for (MGLOfflinePack *pack in packs) {
             if (pack.mbglOfflineRegion->getID() == regionID) {
-                weakSelf.mbglOfflineRegion = pack.mbglOfflineRegion;
-                break;
+                if (strongSelf.mbglOfflineRegion) {
+                    strongSelf->_mbglDatabaseFileSource->setOfflineRegionObserver(*strongSelf.mbglOfflineRegion, nullptr);
+                }
+                strongSelf.mbglOfflineRegion = pack.mbglOfflineRegion;
+                strongSelf->_mbglDatabaseFileSource->setOfflineRegionObserver(*strongSelf.mbglOfflineRegion, std::make_unique<MBGLOfflineDownloadObserver>(strongSelf));
```

Comment:
Since we're swapping out the underlying offline region, it seems like we should also be removing and re-adding the observer.